source/extensions/http/cache/file_system_http_cache/file_system_http_cache.h

```diff
@@ -205,17 +214,67 @@ class FileSystemHttpCache : public HttpCache,
       entries_being_written_ ABSL_GUARDED_BY(cache_mu_);
 
   std::shared_ptr<Common::AsyncFiles::AsyncFileManager> async_file_manager_;
-  CacheStats stats_;
 
+  // Stats and config are held in a shared_ptr so that CacheEvictionThread can use
+  // them even if the cache instance has been deleted while it performed work.
+  std::shared_ptr<CacheShared> shared_;
+
+  CacheEvictionThread& cache_eviction_thread_;
```

Comment:
cache_eviction_thread_ needs a comment that it must be after owner_, since destruction of owner_ could destroy the object to which cache_eviction_thread_ refers.

Reply:
Done (though I'm not sure it's required - it's only after the destructor completes that the class members are destroyed, and at that point you can't use this reference anyway.)

Reply:
Thank you for the correction. My mental pattern matching misfired and I didn't notice.